### PR TITLE
Fix for issue URLDecoder processor should better handle non well form…

### PR DESCRIPTION
**Pull request by jerome73**
Fix for issue #437

### CODE REVIEW

File: logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webAnalytics/URLDecoder.java [hunk 1]
```diff
@@ -104,20 +114,35 @@ private void updateRecord(ProcessContext context, Record record, HashSet<String>
...
+        try {
+            decodedValue = java.net.URLDecoder.decode(value, charset);
+            if (!decodedValue.equals(value)) {
+                final FieldType fieldType = record.getField(fieldName).getType();
+                record.removeField(fieldName);
+                record.setField(fieldName, fieldType, decodedValue);
+            }
+        } catch (IllegalArgumentException e) {
+            if (tryTrick) {
+                value = value.replaceAll("%(?![0-9A-F]{2})", percentEncodedChar);
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webAnalytics/URLDecoder.java]: the regex should be "%(?![0-9a-fA-F]{2})" to be case insensitive.
```suggestion
                value = value.replaceAll("%(?![0-9a-fA-F]{2})", percentEncodedChar);
```

File: logisland-components/logisland-processors/logisland-processor-web-analytics/src/test/java/com/hurence/logisland/processor/webAnalytics/URLDecoderTest.java [hunk 2]
```diff
@@ -176,4 +194,24 @@ public void testNoMatchingField() {
         out.assertFieldEquals("url2", urlVal2);
         out.assertFieldEquals("val4", val4);
     }
+
```
Comment (MiniPlayer) [inline logisland-components/logisland-processors/logisland-processor-web-analytics/src/test/java/com/hurence/logisland/processor/webAnalytics/URLDecoderTest.java]: I guess a unit test with '%ab' and '%AB' would be a good idea ?

Comment (jerome73) [reply logisland-components/logisland-processors/logisland-processor-web-analytics/src/test/java/com/hurence/logisland/processor/webAnalytics/URLDecoderTest.java]: The tests you suggest are already covered (see below)

File: logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webAnalytics/URLDecoder.java (hunk 1 above)
Comment (jerome73) [reply logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webAnalytics/URLDecoder.java]: I agree

**Pull request merged**